code/game/objects/structures/displaycase.dm

```diff
@@ -41,19 +42,16 @@
 	..()
 	take_damage(Proj.get_structure_damage())
 
-/obj/structure/displaycase/take_damage(damage)
-	health -= damage
-	if(health <= 0)
-		if (!destroyed)
-			set_density(0)
-			destroyed = 1
-			new /obj/item/material/shard(loc)
-			for(var/atom/movable/AM in src)
-				AM.dropInto(loc)
-			playsound(src, "shatter", 70, 1)
-			update_icon()
-	else
-		playsound(src.loc, 'sound/effects/Glasshit.ogg', 75, 1)
+/obj/structure/displaycase/destroyed()
+	if(destroyed)
+		return
+	set_density(0)
+	destroyed = TRUE
+	new /obj/item/material/shard(loc)
```

Comment:
Should probably reduce `matter` when you perform this kind of action, unless it's fully qdel'd or something?

Reply:
I don't believe it is qdel'd, this is code I just moved. I'll modify.